docs/agent_metadata_changes.md

```diff
@@ -0,0 +1,45 @@
+# Breaking Change: Updates to Agent DaemonSet Labels, Selectors & Names
+
+Starting in **Operator v1.21.0**, we are introducing changes to the **labels, selectors, and naming conventions** used for the Agent DaemonSets. These updates aim to improve consistency and reduce the length of DaemonSet names managed by [Datadog Agent Profiles (DAP)][1].
+
+These changes may affect **other Kubernetes resources** that rely on **matching these labels**, such as:
+
+- Network Policies
+- Service Mesh configurations
+- Vertical Pod Autoscalers (VPA)
+- Admission Controllers or Mutating Webhooks
+
+If your setup makes any assumptions about the Agent pod labels or DaemonSet names (e.g., for targeting or exclusion), you may need to **update those configurations** to avoid unexpected behavior.
+
+---
+
+## What’s Changing?
+
+| Operator Version | DaemonSet Type | Pod Label Change | Selector Change | DaemonSet Name Change |
```

Comment:
Would it make more sense to put `DaemonSet Type` and `DaemonSet Name Change` next to each other?